magicctapipe/conftest.py

```diff
@@ -0,0 +1,660 @@
+def temp_pandas(tmp_path_factory):
+    return tmp_path_factory.mktemp("pandas")
+
+
+"""
+Local paths
+"""
+
+
+@pytest.fixture(scope="session")
+def base_path():
+    # TO BE CHANGED
+    return Path("/home/elisa/.cache/magicctapipe")
+
+
+@pytest.fixture(scope="session")
+def dl0_gamma_path(base_path):
+    return base_path / "DL0gamma"
+
+
+@pytest.fixture(scope="session")
+def dl0_p_path(base_path):
+    return base_path / "DL0p"
+
+
+@pytest.fixture(scope="session")
+def dl0_M1_path(base_path):
+    return base_path / "DL0M1"
+
+
+@pytest.fixture(scope="session")
+def dl0_M2_path(base_path):
+    return base_path / "DL0M2"
+
+
+@pytest.fixture(scope="session")
+def dl1_lst_path(base_path):
+    return base_path / "DL1LST"
+
+
+@pytest.fixture(scope="session")
+def conf_path(base_path):
+    p = base_path / "config"
+    return p
+
+
+"""
+Custom data
+"""
+
+
+@pytest.fixture(scope="session")
+def dl2_test(temp_DL2_test):
+    """
+    Toy DL2
+    """
+    path = temp_DL2_test / "dl2_test.h5"
+    data = Table()
+    data["obs_id"] = [1, 1, 2, 2, 2, 3, 3]
+    data["event_id"] = [7, 7, 8, 8, 8, 7, 7]
+    data["tel_id"] = [1, 2, 1, 2, 3, 1, 3]
+    data["combo_type"] = [1, 1, 3, 3, 3, 2, 2]
+    data["multiplicity"] = [2, 2, 3, 3, 3, 2, 2]
+    data["timestamp"] = [1, 1, 4, 4, 4, 10, 10]
+    data["pointing_alt"] = [0.6, 0.6, 0.7, 0.7, 0.7, 0.5, 0.5]
+    data["pointing_az"] = [1.5, 1.5, 1.5, 1.5, 1.5, 1.5, 1.5]
+    data["reco_energy"] = [1, 1, 1, 1, 1, 1, 1]
+    data["gammaness"] = [0.5, 0.5, 1, 0.3, 0.5, 1, 1]
+    data["reco_alt"] = [40, 40, 41, 41, 41, 42, 42]
+    data["reco_az"] = [85, 85, 85, 85, 85, 85, 85]
+    write_table_hdf5(
+        data, str(path), "/events/parameters", overwrite=True, serialize_meta=False
+    )
+    return path
+
+
+@pytest.fixture(scope="session")
+def pd_test():
+    """
+    Toy dataframe
+    """
+    df = pd.DataFrame(np.array([[1, 2], [3, 4], [5, 6]]), columns=["a", "b"])
+    return df
+
+
+"""
+Remote paths (to download test files)
+"""
+
+
+@pytest.fixture(scope="session")
+def base_url():
+    return Path("cp02:/fefs/aswg/workspace/elisa.visentin/git_test")
+
+
+@pytest.fixture(scope="session")
+def dl0_gamma_url(base_url):
+    return base_url / "DL0gamma"
+
+
+@pytest.fixture(scope="session")
+def dl0_p_url(base_url):
+    return base_url / "DL0p"
+
+
+@pytest.fixture(scope="session")
+def dl0_M_url(base_url):
+    return base_url / "MAGIC"
+
+
+@pytest.fixture(scope="session")
+def dl1_LST_url(base_url):
+    # BETTER SOLUTION? CHANGED?
+    return base_url / "LST"
+
+
+@pytest.fixture(scope="session")
+def conf_url(base_url):
+    q = base_url / "config.yaml"
+    return q
+
+
+"""
+Downloads: useful functions
+"""
+
+
+@pytest.fixture(scope="session")
+def env_prefix():
+    # ENVIRONMENT VARIABLES TO BE CREATED
+    return "MAGIC_CTA_DATA_"
+
+
+def scp_file(path, url, env_prefix):
+    """
+    Download of test files through rsync
+    """
+    pwd = os.environ[env_prefix + "PASSWORD"]
```

Comment:
you can exchange the ssh keys with the IT cluster, which allows you to login without a password, and thus will avoid putting the password in an environment variable (that is not very secure)

Reply:
This is a temporary solution, before merging it we will have to find a location for the test files (now they are in my workspace, not a good idea because we have to use subprocess.run; if we put them with the MAGIC ones on the same server used my ctapipe-io-magic, we can download them with a dedicated ctapipe function that automatically manages some keys that can be stored as Git Secrets). SSH keys are a big issue because there is no 'test' global user (i.e.: password and username shared by all the Collaboration) and everyone has his keys and his tunnels (and an ssh config file). Public ssh keys can be used to skip passwords, but I don't know if they can be used (again, security issues). Maybe private keys? 
While waiting for data to be stored somewhere, this allows people to test the tests, but for sure we will have to change this. Unfortunately, to access the IT Container I need more than one ProxyJump (and some public and private keys), so that Paramiko can't be used. This is why we must not merge it now :)

Reply:
the keys would work find for individual users, but indeed for implementation in the repository itself for automatic tests this can be a problem
if we can use the same solution as ctapipe_io_magic, this would be preferred.

Nevertheless, the whole pack of files to download is 2 GB, and this is quite some size, plus it makes the processing time long. The largest files are the gamma MCs (2 x 400 MB), and DL1 from LST (1 subrun of 700 MB) It would be good to use dedicated gamma MCs (either ask Georgios to provide two files with e.g. a factor of 30 less events per file,  or maybe  try to cut the file ourselfs. 
Similar with the data file - to find a much smaller LST subrun we can always take the last subrun of a given run.

Reply:
As for MCs, for sure we can ask for smaller datasets (I can try to ask Georgios, maybe he won't curse me) while waiting for the permissions needed to upload them to the MAGIC server.
As for the LST sub-run, I will have to do some tests because there must be some coincident events ('a lot' because they have to survive quality cuts up to DL3), and the wobble shifts are not synchronous... 
I don't know if we can cut the files (w/o running into exceptions or errors), but I can take a look at this, too.

Reply:
no need for the `scp`/`rsync` function anymore, we use now the ctapipe `download_file_cached` function, after the data have been moved to the MAGIC webserver (btw, also this is temporary, possibly we will put them at PIC).